v2/kafka-common/src/main/java/com/google/cloud/teleport/v2/kafka/coders/GenericRecordCoder.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.kafka.coders;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.concurrent.ExecutionException;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.beam.sdk.coders.AtomicCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.extensions.avro.coders.AvroCoder;
+
+/**
+ * An {@link AtomicCoder} for {@link GenericRecord}.
+ *
+ * <p>This coder is used when the schema of the incoming {@link GenericRecord} can only be known at
+ * runtime.
+ */
+public class GenericRecordCoder extends AtomicCoder<GenericRecord> {
```

Comment:
This seems like the same class as inhttps://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1448/files#diff-6baa0b4edc19ab903f81821ba100085478a34a74323aba6bd6e66899dba6190c.
But in #1448 it's in the common module, not kafka-common. I think common module is a better place because this coder is not kafka-specific.
OK to refactor this later in a separate PR, so just FYI.

Reply:
Done